File: src/types/output_parameter.ts
import { FuzzyOutputValueResponse } from "./fuzzy_output_value";

export type CreateOutputParameterRequest = {
    problem_id: number;
}

export type UpdateOutputParameterRequest = {
    name: string;
    start: number;
    end: number;
}

export type OutputParameterResponse = {
    id: number;
    problem_id: number;
    name: string;
    start: number;
    end: number;
    fuzzy_output_values: Array<FuzzyOutputValueResponse>;
}

File: src-tauri/src/domain/entities/output_parameter.rs
use serde::{Deserialize, Serialize};

use crate::domain::entities::fuzzy_output_value::FuzzyOutputValue;

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct OutputParameter {
    pub id: i64,
    pub problem_id: i64,
    pub name: String,
    pub start: f32,
    pub end: f32,
    pub fuzzy_output_values: Vec<FuzzyOutputValue>,
}
